aviary/interface/methods_for_level2.py

```diff
@@ -450,6 +487,57 @@ def check_and_preprocess_inputs(self):
         # Fill in anything missing in the options with computed defaults.
         preprocess_crewpayload(self.aviary_inputs)
 
+        if self.mission_method is HEIGHT_ENERGY:
+            self.phase_separator()
+
+        # Target_distance verification for all phases
+        # Checks to make sure target_distance is positive,
+        for idx, phase_name in enumerate(self.phase_info):
+            if 'user_options' in self.phase_info[phase_name]:
+                if 'target_distance' in self.phase_info[phase_name]["user_options"]:
+                    target_distance = self.phase_info[phase_name]["user_options"]["target_distance"]
+                    if target_distance[0] <= 0:
+                        raise ValueError(
+                            f"Invalid target_distance in [{phase_name}].[user_options]. "
+                            f"Current (value: {target_distance[0]}), (units: {target_distance[1]}) <= 0")
+
+        # Target_time verification for all phases
+        # Checks to make sure target_time is positive,
+        # duration_bounds and initial_guesses for time have not been set,
+        # Then sets duration_bounds, initial_guesses, and fixed_duration
+        for idx, phase_name in enumerate(self.phase_info):
+            if 'user_options' in self.phase_info[phase_name]:
+                if 'target_time' in self.phase_info[phase_name]["user_options"]:
```

Comment:
I just noticed that we already have "fix_duration" in the phase info.  I think we can get rid of "target_time", and just require the user to set the "target" time as the "times" in "initial_guesses".

Also, given that fix_duration is already there, you might not have to do any other processing, since something should already be in aviary (though whether it works is a question).

Reply:
we're not going to use fix_duration because it only allows a true/false flag whereas I want to allow the user to directly specify a value and units for that duration in a single place. And if `target_duration` is true, then that also sets the `fix_duration` flag.